spec/models/piece_spec.rb

```diff
@@ -55,4 +55,20 @@
       expect(piece.diff(3, piece.x_position)).to eq 3
     end
   end
+
+  describe '.move_to!' do
+    it 'should move the piece to the destination square' do
+      expect(piece.move_to!(2, 6)).to be true
+    end
+
+    it 'return true if the move is valid' do
+      expect(piece.valid_move?(4, 7)).to eq true
+    end
+  end
+
+  describe '.capture_piece' do
+    it 'returns false is the destination is occupied' do
```

Comment:
For this test, you'll need some dummy pieces created first, so the before block on lines 8-12 could be moved outside of the `.obstructed?` describe block.